### DISCUSSION TITLE
HasManyThrough in a form?

### DISCUSSION THREAD

**Discussion by StaffCollab** [filamentphp/filament#6598] [Help] [answered]
Hello everyone.

I have a HasManyThrough relationship. It is a payment plan that has many ads, and ads have many items.
Now, I want to display a list of items (which is actually products in the order), when creating a new payment plan, the question is how do I show a HasManyThrough relationship in a form? The only way I see in the docs to use a hasManyThrough is through a relation manager, which is not a solution for a form.

Please, can anyone guide me on that?
Thanks a million.
Yakov

**Comment by danharrin** [comment, marked as answer]
You could use a Repeater `relationship('ads')` to add ads, and then a Repeater `relationship('items')` inside that to add items

**Reply by danswiser** [reply]
@danharrin,

Could this be done in any way without having the repeaters nested? Since each model of the relationship has a lot of properties to add I have the parent repeater and child repeater in separate steps of a Wizard for usability.

**Reply by Aero-Mohamed** [reply]
same problem here! any ideas?

**Reply by StaffCollab** [reply]
You can do it through a repeater without a relationship method; instead, use a default and query the relationship manually.
But I think you might need to save it manually as well since it is not a relationship repeater.

Or if you need to set the data after something happens in the form, you can use a `$set()` on it that's my code
~~~
  Step::make('Payment Plan')
                        ->schema([
                         //columns
                        ])->columns(2)
                        ->afterValidation(function (Get $get, Set $set, Livewire $livewire) {
                            $livewire->calculateParticulars($get);
                            $set('particulars', $this->particulars);
                        }),
 Step::make('Payment Method')
                        ->schema([
                            Repeater::make('particulars')
                                ->schema([
                                    DatePicker::make('payment_date')
                                        ->native(false),
                                    TextInput::make('payment_amount')
                                ])
                                ->columns(2)
                                ->grid(2)
                                ->addable(false)
                                ->reorderable(false)
                                ->deletable(false), 
~~~

**Comment by classicalguss** [comment]
I don't understand what's the  issue here. I changed the source code to make the getRelationship method accept HasManyThrough and it seems to work. Am I missing something? I'm thinking of making a PR, but I'm worried it might cause an issue that I may not be aware of?

**Reply by danharrin** [reply]
Does it work for creating, editing and deleting items, or just viewing them?

**Reply by classicalguss** [reply]
Seems to work. 

I tried it on this fresh laravel installation https://github.com/classicalguss/testing-filament-repeater

And tested it on Laravel's HasManyThrough example in the documentation as outlined here 
https://laravel.com/docs/11.x/eloquent-relationships#has-many-through
(Project -> Environment -> Deployments)

Here is a small recording as POC. I edited a deployment, and then deleted it.
https://github.com/user-attachments/assets/c360bc1a-63c5-4eb4-833c-40890ca56ade

**Reply by danharrin** [reply]
Feel free to make a PR
